cartographer/sensor/range_data_test.cc

```diff
@@ -15,45 +15,139 @@
 
 namespace cartographer {
 namespace sensor {
 namespace {
 
 using ::testing::Contains;
+using ::testing::FloatNear;
 using ::testing::PrintToString;
+constexpr float kPrecision = 0.001f;
 
-// Custom matcher for Eigen::Vector3f entries.
+// TODO (brandon-northcutt): place this function in a common location with
+// identical function in compressed_point_cloud_test.cc
 MATCHER_P(ApproximatelyEquals, expected,
           string("is equal to ") + PrintToString(expected)) {
-  return (arg - expected).isZero(0.001f);
+  return (arg - expected).isZero(kPrecision);
 }
 
-TEST(RangeDataTest, Compression) {
-  const std::vector<Eigen::Vector3f> returns = {Eigen::Vector3f(0, 1, 2),
-                                                Eigen::Vector3f(4, 5, 6),
-                                                Eigen::Vector3f(0, 1, 2)};
-  const RangeData range_data = {
-      Eigen::Vector3f(1, 1, 1), returns, {Eigen::Vector3f(7, 8, 9)}};
+class RangeDataTest : public ::testing::Test {
+ protected:
+  RangeDataTest() {
+    origin_ = Eigen::Vector3f(1, 1, 1);
+    returns_.emplace_back(0, 1, 2);
+    returns_.emplace_back(4, 5, 6);
+    returns_.emplace_back(0, 1, 2);
+    misses_.emplace_back(7, 8, 9);
+  }
+  void RangeDataToProtoTest() {}
+
+  void CompressedRangeDataToProtoTest() {
+    const CompressedPointCloud compressed_returns(returns_);
+    const CompressedPointCloud compressed_misses(misses_);
+    const CompressedRangeData compressed_range_data = {
+        origin_, compressed_returns, compressed_misses
+    };
+    const auto proto = ToProto(compressed_range_data);
+  }
+  Eigen::Vector3f origin_;
+  std::vector<Eigen::Vector3f> returns_;
+  std::vector<Eigen::Vector3f> misses_;
+};
+
+TEST_F(RangeDataTest, Compression) {
+  const RangeData range_data = {origin_, returns_, misses_};
   const RangeData actual = Decompress(Compress(range_data));
-  EXPECT_TRUE(actual.origin.isApprox(Eigen::Vector3f(1, 1, 1), 1e-6));
+  EXPECT_TRUE(actual.origin.isApprox(origin_, 1e-6));
   EXPECT_EQ(3, actual.returns.size());
   EXPECT_EQ(1, actual.misses.size());
-  EXPECT_TRUE(actual.misses[0].isApprox(Eigen::Vector3f(7, 8, 9), 0.001f));
+  EXPECT_TRUE(actual.misses[0].isApprox(misses_[0], 0.001f));
 
-  // Returns will be reordered, so we compare in an unordered manner.
-  EXPECT_EQ(3, actual.returns.size());
-  EXPECT_THAT(actual.returns,
-              Contains(ApproximatelyEquals(Eigen::Vector3f(0, 1, 2))));
-  EXPECT_THAT(actual.returns,
-              Contains(ApproximatelyEquals(Eigen::Vector3f(4, 5, 6))));
+  // Returns may be reordered, so we compare in an unordered manner.
+  for (uint i = 0; i < returns_.size(); ++i) {
+    EXPECT_THAT(actual.returns, Contains(ApproximatelyEquals(returns_[i])));
+  }
+  for (uint i = 0; i < misses_.size(); ++i) {
+    EXPECT_THAT(actual.misses, Contains(ApproximatelyEquals(misses_[i])));
+  }
+}
+
+TEST_F(RangeDataTest, RangeDataToProto) {
+  const auto proto = ToProto(RangeData{origin_, returns_, misses_});
```

Comment:
This unit test to me does not increase confidence in the code. How about Converting to a proto and back and check for equality? Similar for the compressed range data: Extract a function to compare unordered and with limited precision and reuse it for the compressed range data test?